### PR TITLE
Add several dataset loaders

### PR DESCRIPTION
**Pull request by xinyuejohn**
**PR Checklist**

-   [ ] This comment contains a description of changes (with reason)
-   [ ] Referenced issue is linked
-   [ ] If you've fixed a bug or added code that should be tested, add tests!
-   [ ] Documentation in `docs` is updated

**Description of changes**
Changed ehrapy/data/_datasets.py.

**Technical details**
Added several dataset loaders.

**Additional context**

### CODE REVIEW

**Review by Zethson — APPROVED** [review]
LGTM

**Comment by Zethson** [comment]
@xinyuejohn but nox is still failing. Please fix everything (ignore the Numpy security thing)

File: ehrapy/data/_datasets.py [hunk 1]
```diff
@@ -181,3 +181,237 @@ def chronic_kidney_disease(
...
+        backup_url="https://figshare.com/ndownloader/files/34179327",
+        columns_obs_only=columns_obs_only,
+        extension="csv",
+        index_column="patient_id",
+    )
+
+    return adata
+
+
+def thyroid(name: str, encoded: bool = False) -> AnnData:
```
Comment (Zethson) [inline ehrapy/data/_datasets.py:384]: Oh, wait. I think this should always return the full dataset by default. Not a subset.

File: ehrapy/data/_datasets.py [hunk 2]
```diff
@@ -181,3 +181,237 @@ def chronic_kidney_disease(
...
+    columns_obs_only = ["dataset_name"]
+    adata: AnnData = read(
+        dataset_path=f"{ehrapy_settings.datasetdir}/thyroid.csv",
+        download_dataset_name="thyroid.csv",
+        backup_url="https://figshare.com/ndownloader/files/34179333",
+        columns_obs_only=columns_obs_only,
+        extension="csv",
+        index_column="patient_id",
+    )
+    adata = adata[adata.obs["dataset_name"] == name].copy()
```
Comment (Zethson) [inline ehrapy/data/_datasets.py:413]: No need to filter here

File: ehrapy/data/_datasets.py (hunk 1 above)
Comment (xinyuejohn) [reply ehrapy/data/_datasets.py]: No problem, already fixed.

File: ehrapy/data/_datasets.py (hunk 2 above)
Comment (xinyuejohn) [reply ehrapy/data/_datasets.py]: Fixed.

**Pull request merged**